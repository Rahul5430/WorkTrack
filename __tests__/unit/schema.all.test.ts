describe('All schemas load and snapshot', () => {
	it('loads main app schema and matches snapshot', () => {
		const schema = require('../../src/db/watermelon/schema').default;
		expect(schema.tables.length).toBeGreaterThan(0);
		expect(schema).toMatchSnapshot({ version: expect.any(Number) });
	});
});
