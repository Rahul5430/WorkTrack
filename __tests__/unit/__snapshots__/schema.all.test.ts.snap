// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All schemas load and snapshot loads main app schema and matches snapshot 1`] = `
{
  "tables": [
    {
      "columns": [
        {
          "isIndexed": true,
          "name": "date",
          "type": "string",
        },
        {
          "name": "status",
          "type": "string",
        },
        {
          "name": "created_at",
          "type": "number",
        },
        {
          "name": "last_modified",
          "type": "number",
        },
        {
          "name": "is_advisory",
          "type": "boolean",
        },
        {
          "isIndexed": true,
          "name": "tracker_id",
          "type": "string",
        },
        {
          "name": "needs_sync",
          "type": "boolean",
        },
        {
          "isOptional": true,
          "name": "sync_error",
          "type": "string",
        },
        {
          "isOptional": true,
          "name": "retry_count",
          "type": "number",
        },
      ],
      "name": "work_tracks",
    },
    {
      "columns": [
        {
          "name": "owner_id",
          "type": "string",
        },
        {
          "name": "name",
          "type": "string",
        },
        {
          "name": "color",
          "type": "string",
        },
        {
          "name": "created_at",
          "type": "number",
        },
        {
          "name": "is_default",
          "type": "boolean",
        },
        {
          "name": "tracker_type",
          "type": "string",
        },
      ],
      "name": "trackers",
    },
    {
      "columns": [
        {
          "name": "tracker_id",
          "type": "string",
        },
        {
          "name": "shared_with",
          "type": "string",
        },
        {
          "name": "permission",
          "type": "string",
        },
        {
          "name": "created_at",
          "type": "number",
        },
      ],
      "name": "shared_trackers",
    },
  ],
  "version": Any<Number>,
}
`;
